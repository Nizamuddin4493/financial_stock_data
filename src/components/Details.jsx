import React from 'react';
import { Link } from 'react-router-dom';
import stock_image from './images/stock-market.jpg'

const Details = () => (
  <>
    <div>
      <div className="detals">
      <Link to="/" className="router_link"><h3 className="details_back_arrow"> &#8617; Back</h3></Link>
        <h2>NasDaq</h2>
        <div className="details_data">
          <div>
            <div className="details_info"><h4>NasDaq:</h4><p>1275721002015</p></div>
            <div className="details_info"><h4>NasDaq:</h4><p>1275721002015</p></div>
            <div className="details_info"><h4>NasDaq:</h4><p>1275721002015</p></div>
          </div>
          <div>
            <div className="details_info"><h4>NasDaq:</h4><p>1275721002015</p></div>
            <div className="details_info"><h4>NasDaq:</h4><p>1275721002015</p></div>
            <div className="details_info"><h4>NasDaq:</h4><p>1275721002015</p></div>
          </div>
          <div>
            <div className="details_info"><h4>NasDaq:</h4><p>1275721002015</p></div>
            <div className="details_info"><h4>NasDaq:</h4><p>1275721002015</p></div>
            <div className="details_info"><h4>NasDaq:</h4><p>1275721002015</p></div>
          </div>
        </div>
        <img src={ stock_image } alt="Share Exchange" />
      </div>
    </div>
  </>
);


export default Details;