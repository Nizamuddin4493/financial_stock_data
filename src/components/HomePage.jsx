import React from 'react';
import { Route, Routes } from 'react-router';
import Categories from './Categories';
import Details from './Details';
import Header from './Header';
import SearchForm from './SearchForm';

function HomePage() {
  return (
    <>
      <div className="homePageDesign">
        <Header />
        <hr />
        <SearchForm />
        <Routes>
          <Route path="/" element={<Categories />} />
          <Route path="/Details" element={<Details />} />
        </Routes>
      </div>
    </>
  );
}

export default HomePage;
