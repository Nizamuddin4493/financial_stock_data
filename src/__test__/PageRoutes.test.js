import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import HomePage from '../components/HomePage';
import Details from '../components/Details';

const HomeProvider = () => (
  <Provider store={store}>
    <Router>
      <HomePage />
    </Router>
  </Provider>
);

const DetailsProvider = () => (
  <Provider store={store}>
    <Router>
      <Details />
    </Router>
  </Provider>
);

describe('UI is rendered properly', () => {
  it('renders the home page', () => {
    const homePage = renderer.create(<HomeProvider />).toTree();
    expect(homePage).toMatchSnapshot();
  });

  it('renders the details page of companies properly', () => {
    const detailsPage = renderer.create(<DetailsProvider />).toTree();
    expect(detailsPage).toMatchSnapshot();
  });
});
