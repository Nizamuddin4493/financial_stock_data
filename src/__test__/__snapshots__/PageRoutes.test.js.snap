// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI is rendered properly renders the details page of companies properly 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {},
  "rendered": Object {
    "instance": null,
    "nodeType": "component",
    "props": Object {
      "children": <BrowserRouter>
        <Details />
      </BrowserRouter>,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "rendered": Object {
      "instance": null,
      "nodeType": "component",
      "props": Object {
        "children": <Details />,
      },
      "rendered": Object {
        "instance": null,
        "nodeType": "component",
        "props": Object {
          "basename": undefined,
          "children": <Details />,
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "navigationType": "POP",
          "navigator": Object {
            "action": "POP",
            "back": [Function],
            "block": [Function],
            "createHref": [Function],
            "forward": [Function],
            "go": [Function],
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "default",
              "pathname": "/",
              "search": "",
              "state": null,
            },
            "push": [Function],
            "replace": [Function],
          },
        },
        "rendered": Object {
          "instance": null,
          "nodeType": "component",
          "props": Object {},
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  "type": [Function],
}
`;

exports[`UI is rendered properly renders the home page 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {},
  "rendered": Object {
    "instance": null,
    "nodeType": "component",
    "props": Object {
      "children": <BrowserRouter>
        <HomePage />
      </BrowserRouter>,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "rendered": Object {
      "instance": null,
      "nodeType": "component",
      "props": Object {
        "children": <HomePage />,
      },
      "rendered": Object {
        "instance": null,
        "nodeType": "component",
        "props": Object {
          "basename": undefined,
          "children": <HomePage />,
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "navigationType": "POP",
          "navigator": Object {
            "action": "POP",
            "back": [Function],
            "block": [Function],
            "createHref": [Function],
            "forward": [Function],
            "go": [Function],
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "default",
              "pathname": "/",
              "search": "",
              "state": null,
            },
            "push": [Function],
            "replace": [Function],
          },
        },
        "rendered": Object {
          "instance": null,
          "nodeType": "component",
          "props": Object {},
          "rendered": Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Header />,
                <hr />,
                <SearchForm />,
                <Routes>
                  <Route
                    element={<Categories />}
                    path="/"
                  />
                  <Route
                    element={<Details />}
                    path="/:id"
                  />
                </Routes>,
              ],
              "className": "homePageDesign",
            },
            "rendered": Array [
              Object {
                "instance": null,
                "nodeType": "component",
                "props": Object {},
                "rendered": Object {
                  "instance": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h1>
                      Financial Stock Data
                    </h1>,
                    "className": "header",
                  },
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Financial Stock Data",
                      },
                      "rendered": Array [
                        "Financial Stock Data",
                      ],
                      "type": "h1",
                    },
                  ],
                  "type": "div",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "nodeType": "host",
                "props": Object {},
                "rendered": Array [],
                "type": "hr",
              },
              Object {
                "instance": null,
                "nodeType": "component",
                "props": Object {},
                "rendered": Object {
                  "instance": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": <form
                      className="searchForm"
                    >
                      <input
                        placeholder="Search..."
                        type="text"
                      />
                    </form>,
                  },
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": <input
                          placeholder="Search..."
                          type="text"
                        />,
                        "className": "searchForm",
                      },
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "nodeType": "host",
                          "props": Object {
                            "placeholder": "Search...",
                            "type": "text",
                          },
                          "rendered": Array [],
                          "type": "input",
                        },
                      ],
                      "type": "form",
                    },
                  ],
                  "type": "div",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "nodeType": "component",
                "props": Object {
                  "children": Array [
                    <Route
                      element={<Categories />}
                      path="/"
                    />,
                    <Route
                      element={<Details />}
                      path="/:id"
                    />,
                  ],
                },
                "rendered": Object {
                  "instance": null,
                  "nodeType": "component",
                  "props": Object {},
                  "rendered": Object {
                    "instance": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [],
                      "className": "categories",
                    },
                    "rendered": Array [],
                    "type": "div",
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  "type": [Function],
}
`;
